src/Sarif/ConsoleLogger.cs

```diff
@@ -291,7 +291,7 @@ private string FormatNotificationMessage(Notification notification)
                 sb.Append(notification.RuleId + " : ");
             }
 
-            sb.Append(notification.Message);
+            sb.Append(notification.Message.Text);
```

Comment:
Existing bug that went unnoticed when we introduced the `Message` object. I noticed this when I ran the Multitool and it failed with an `ExecutionEngine` error, but the output didn't tell me what had gone wrong (it was a file not found error). The output just said "Microsoft.CodeAnalysis.Sarif.Message".

Reply:
Nuts. You see the subtlety here? sb.Append silently fell back from calling sb.Append(params string[]) to sb.AppendLine(params object[]) when the type changed. :) AppendLine calls Message.ToString() which by default just emits its fully qualified type name. We'd need an analyze that bans all calls to String.Concat(object)/StringBuilder.Append(object) to find this thing. 

Or author a doggone unit test! (which would have caught this specific case) :)

---
In reply to: [205180107](https://github.com/Microsoft/sarif-sdk/pull/968#discussion_r205180107) [](ancestors = 205180107)

Reply:
Oh yes, the failure mode was quite clear.

---
In reply to: [205192274](https://github.com/Microsoft/sarif-sdk/pull/968#discussion_r205192274) [](ancestors = 205192274,205180107)